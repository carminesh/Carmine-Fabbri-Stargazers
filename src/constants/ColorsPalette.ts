export default {
    LIGHT_BACKGROUND: '#FFFFFF',
    BLUE_ACCENT: '#2C68F7',
    BLACK_ACCENT: '#333333',
    BLACK: '#000000',
    BLACK90: '#1c1c1e',
    BLACK75: '#2c2c2e',
    BLACK50: '#3a3a3c',
    BLACK25: '#48484a',
    BLACK10: '#636366',
    GRAY: '#a6a6a6',
    LIGHT_GRAY: '#f5f5f5',
    WHITE10: '#aeaeb2',
    WHITE25: '#c7c7cc',
    WHITE50: '#d1d1d6',
    WHITE75: '#e5e5ea',
    WHITE90: '#f2f2f7',
    WHITE: '#FFFFFF',
};
