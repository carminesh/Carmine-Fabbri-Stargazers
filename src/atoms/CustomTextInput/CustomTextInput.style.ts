import { ScaledSize, StyleSheet } from 'react-native';
import ColorsPalette from '../../constants/ColorsPalette';

export const style = (dimensions: ScaledSize) =>
    StyleSheet.create({
        textInputContainer: {
            width: dimensions.width - 32,
            height: 45,
            backgroundColor: ColorsPalette.LIGHT_BACKGROUND,
            borderRadius: 5,
            borderWidth: 1,
            borderColor: ColorsPalette.GRAY,
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center',
            paddingHorizontal: 20,
        },
        icon: {
            tintColor: ColorsPalette.GRAY,
            marginRight: 10,
            width: 15,
            height: 15,
        },
        inputTextField: {
            width: '100%',
            height: '70%',
        },
    });
