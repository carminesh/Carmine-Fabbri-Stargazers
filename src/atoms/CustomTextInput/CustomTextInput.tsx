import React, { useCallback, useState } from 'react';
import { Image, ImageSourcePropType, TextInput, useWindowDimensions, View } from 'react-native';
import ColorsPalette from '../../constants/ColorsPalette';
import { style } from './CustomTextInput.style';

interface CustomTextInputProps {
    testID?: string;
    icon?: ImageSourcePropType;
    placeholder: string;
    textValue: string;
    onChangeText: React.Dispatch<React.SetStateAction<string>>;
}

const CustomTextInput: React.FC<CustomTextInputProps> = ({ testID, icon, placeholder, textValue, onChangeText, ...props }) => {
    const dimensions = useWindowDimensions();
    const styles = style(dimensions);

    const [isFocused, setIsFocused] = useState<boolean>(false);

    return (
        <View testID="text-input-container" style={[styles.textInputContainer, isFocused && { borderColor: ColorsPalette.BLUE_ACCENT }]}>
            {icon && <Image testID="custom-text-input-icon" style={[styles.icon, isFocused && { tintColor: ColorsPalette.BLUE_ACCENT }]} source={icon} resizeMode="contain" />}
            <TextInput
                testID={testID ? testID : 'custom-text-input'}
                style={styles.inputTextField}
                placeholder={placeholder}
                onChangeText={(text) => onChangeText(text)}
                value={textValue}
                onFocus={() => setIsFocused(true)}
                onBlur={() => setIsFocused(false)}
            />
        </View>
    );
};

export default CustomTextInput;
