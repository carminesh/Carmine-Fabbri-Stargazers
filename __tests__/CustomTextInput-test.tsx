import 'react-native';
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import '@testing-library/jest-native/extend-expect';
import CustomTextInput from '../src/atoms/CustomTextInput/CustomTextInput';
import renderer from 'react-test-renderer';
import ColorsPalette from '../src/constants/ColorsPalette';

describe('CustomTextInput testing', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<CustomTextInput placeholder="Search" textValue="" onChangeText={() => {}} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should change the text value when the input field is changed', async () => {
        const textValue = 'initial value';
        const onChangeText = jest.fn();
        const { getByTestId } = render(<CustomTextInput placeholder="Search" textValue={textValue} onChangeText={onChangeText} />);

        const inputField = getByTestId('custom-text-input');
        fireEvent.changeText(inputField, 'new value');
        expect(onChangeText).toHaveBeenCalledWith('new value');
    });

    it('should not have a border color of blue when the input field is not focused', async () => {
        const textValue = '';
        const onChangeText = jest.fn();
        const { getByTestId } = render(<CustomTextInput placeholder="Search" textValue={textValue} onChangeText={onChangeText} />);

        const textInputContainer = getByTestId('text-input-container');
        expect(textInputContainer).toHaveStyle({ borderColor: ColorsPalette.GRAY });
    });

    it('should display the icon passed as a prop', async () => {
        const icon = { uri: 'https://example.com/icon.png' };
        const { queryByTestId } = render(<CustomTextInput placeholder="Search" textValue="" onChangeText={() => {}} icon={icon} />);

        const iconImage = queryByTestId('custom-text-input-icon');
        expect(iconImage).toBeTruthy();
    });

    it('should not display an icon if not passed as a prop', async () => {
        const { queryByTestId } = render(<CustomTextInput placeholder="Search" textValue="" onChangeText={() => {}} />);

        const iconImage = queryByTestId('custom-text-input-icon');
        expect(iconImage).toBeFalsy();
    });
});
