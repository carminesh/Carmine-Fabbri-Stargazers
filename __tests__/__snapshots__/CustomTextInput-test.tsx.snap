// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextInput testing renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#a6a6a6",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": 718,
      },
      false,
    ]
  }
  testID="text-input-container"
>
  <TextInput
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="Search"
    style={
      {
        "height": "95%",
        "width": "100%",
      }
    }
    testID="custom-text-input"
    value=""
  />
</View>
`;
